src/core/ext/filters/client_channel/lb_policy/pick_first/pick_first.cc

```diff
@@ -506,11 +508,13 @@ static void pf_connectivity_changed_locked(grpc_exec_ctx* exec_ctx, void* arg,
         }
         sd->curr_connectivity_state =
```

Comment:
What happens if we simply remove the call to `grpc_subchannel_check_connectivity()` and instead unconditionally call `grpc_lb_subchannel_data_start_connectivity_watch()`?  Does anything break?

Reply:
Excellent suggestion. I've also removed that pesky loop around the switch.